tests/test_recruiters.py

```diff
@@ -954,6 +963,7 @@ def recruiter(
             mocks["get_base_url"].return_value = "http://fake-domain"
             mocks["os"].getenv.return_value = "fake-host-domain"
```

Comment:
Probably a good idea to add a test somewhere that when `publish_experiment=False` and you call `open_recruitment` then `publish_study` is not called, but when `publish_experiment=True` it is called?

Reply:
Added two tests